### PR TITLE
ht_5/zaikouskaya

### PR DESCRIPTION
**Pull request by KateZaikouskaya**

### CODE REVIEW

File: app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        if (a.length()>50 || b.length()>50)
+            throw new IllegalArgumentException("Strings are too long");
+
+        //checking strings for non-alphabetic characters
+        Pattern pattern = Pattern.compile("[.[^A-Za-z]]+");
+        Matcher matcherA = pattern.matcher(a);
+        Matcher matcherB = pattern.matcher(b);
+
+        if (matcherA.find() || matcherB.find())
+            throw new IllegalArgumentException("Strings contain not only letters");
```
Comment (AlexOreshkevich) [inline app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java:22]: This is not covered by tests

File: app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AnagramUtils {
+    public static boolean isAnagram(@NotNull String a, @NotNull String b){
+
+        //checking string lengths
+        if (a.length()>50 || b.length()>50)
```
Comment (AlexOreshkevich) [inline app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java:13]: code style

File: app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package by.academy.jc.zaikouskaya;
+
+import org.jetbrains.annotations.NotNull;
```
Comment (AlexOreshkevich) [inline app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java:3]: Use `javax.validation.NotNull` instead please.

See Joshua Bloch programming through interface,

**Review by AlexOreshkevich — APPROVED** [review]
graded

File: app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java (hunk 3 above)
Comment (KateZaikouskaya) [reply app/src/main/java/by/academy/jc/zaikouskaya/AnagramUtils.java:3]: My IDE can't find this library.

**Pull request closed without merging**